src/Config/Settings.ts

```diff
@@ -33,8 +33,31 @@ export const TASK_FORMATS = {
 
 export type TASK_FORMATS = typeof TASK_FORMATS; // For convenience to make some typing easier
 
+export class GlobalFilter {
```

Comment:
We haven't talked about this before, but the general policy of this project is to have one significant class per source file. So this can this go in its own file please.

This also makes it easy to write and find its tests in a file of matching name.